perf_workflow/run_perf_suite/ccr_perf_test.py

```diff
@@ -0,0 +1,294 @@
+                        args = PerfSuiteArgs(
+                            self.test_config["Workload"], workload_options, 1, 0
+                        )
+                        try:
+                            logging.info(
+                                f"Starting test {self.description}, Leader: {leaderCluster.endpoint}, Follower: {followerCluster.endpoint}"
+                            )
+                            # Execute the suite.
+                            perf_test_suite = PerfTestSuite(
+                                self.bundle_manifest,
+                                target_hosts,
+                                self.security,
+                                current_workspace,
+                                tests_dir,
+                                args,
+                                self.TEST_OWNER,
+                                self.CCR_SCENARIO,
+                            )
+                            perf_test_suite.execute()
+
+                            # Get test results and validate the metrics.
+                            result_json_file_path = max(
+                                glob.glob(tests_dir + "/*.json"), key=os.path.getmtime
+                            )
+                            with open(result_json_file_path, "r") as file:
+                                self.verify_result(json.load(file), leaderCluster, followerCluster)
+                        except Exception as e:
+                            self.failure_reason = str(e)
+                            logging.error(f"Test failed for {self.description}: {e}")
+                            traceback.print_exc()
+
+    def verify_result(self, result, leaderCluster, followerCluster):
+        self.verify_checkpoints(followerCluster)
+        self.verify_doc_count(leaderCluster, followerCluster)
+        self.execution_id = result["testExecutionId"]
+        for telemetry_result in result["testResults"]["customTelemetryData"]["overall"]:
+            if (
+                telemetry_result["telemetryDevice"] == "ccr-stats"
+                and telemetry_result["metric"] == "replication_lag"
+            ):
+                # TODO: Add support for comparison with benchmark run
+                assert (
+                    telemetry_result["p100"] > 0
+                ), "Replication lag should be non-zero"
+                assert (
+                    telemetry_result["p100"] < self.MAX_ALLOWED_REPLICATION_LAG_SECONDS
+                ), "Replication lag p100 was greater than a minute"
+                self.test_succeeded = True
+                logging.info(f"Test succeeded for {self.description}")
+
+    def verify_doc_count(self, leaderCluster, followerCluster):
+        index_name = self.test_config["Workload"]
+        path = f"/{index_name}/_count?format=json"
+
+        leader_url = "".join([leaderCluster.endpoint_with_port(), path])
+        leader_resp = requests.get(url=leader_url, auth=self.CREDS, verify=False)
```

Comment:
Where is `CREDS` defined?

Reply:
My bad. Got removed accidentally. Added it back now.